File: iOS5bookExamples/ch25p708timeConsumingNavigation/p598p614timeConsumingNavigation/MyTableViewCell.m


#import "MyTableViewCell.h"
#import <QuartzCore/QuartzCore.h>

@implementation MyTableViewCell

- (void)setSelected:(BOOL)selected animated:(BOOL)animated {
    [super setSelected:selected animated:animated]; // note order of operations
    if (selected) {
        UIActivityIndicatorView* v = 
        [[UIActivityIndicatorView alloc] 
         initWithActivityIndicatorStyle:UIActivityIndicatorViewStyleWhiteLarge];
        v.center = 
        CGPointMake(self.bounds.size.width/2.0, self.bounds.size.height/2.0);
        v.frame = CGRectIntegral(v.frame);
        v.color = [UIColor yellowColor];
        v.tag = 1001;
        [self.contentView addSubview:v];
        [v startAnimating];
    } else {
        [[self.contentView viewWithTag:1001] removeFromSuperview];
        // no harm if nonexistent
    }
}

@end

File: iOS9bookExamples/bk2ch12p564timeConsumingNavigation/ch25p837timeConsumingNavigation/MyTableViewCell.swift


import UIKit


class MyTableViewCell: UITableViewCell {

    override func setSelected(selected: Bool, animated: Bool) {
        
        if selected {
            let v = UIActivityIndicatorView(activityIndicatorStyle:.WhiteLarge)
            v.color = UIColor.yellowColor()
            dispatch_async(dispatch_get_main_queue()) {
                v.backgroundColor = UIColor(white:0.2, alpha:0.6)
            }
            v.layer.cornerRadius = 10
            v.frame = v.frame.insetBy(dx: -10, dy: -10)
            let cf = self.contentView.convertRect(self.bounds, fromView:self)
            v.center = CGPointMake(cf.midX, cf.midY);
            v.frame.makeIntegralInPlace() // ?
            v.tag = 1001
            self.contentView.addSubview(v)
            v.startAnimating()

        } else {
            if let v = self.viewWithTag(1001) {
                v.removeFromSuperview()
            }
        }
        super.setSelected(selected, animated: animated)

    }

}
